flake.nix

```diff
@@ -33,9 +35,31 @@
       # Use the expression from Nixpkgs instead of rewriting it here.
       sile = pkgs.sile.overrideAttrs(oldAttr: rec {
         version = "${(pkgs.lib.importJSON ./package.json).version}-${version_rev}-flake";
-        src = builtins.filterSource
-          (path: type: type != "directory" || baseNameOf path != ".git")
-        ./.;
+        src = nix-filter.lib {
+          root = ./.;
+          exclude = [
+            # Nix files
+            "flake.nix"
```

Comment:
Shouldn't changes to the `*.nix` files be *included* in the sources considered? Changing them should cause rebuilds right?

Reply:
That's a good question. When Nix "builds" the derivation of the source directory, it already _evaluated_ all the Nix files it had to evaluate, so it doesn't need any Nix file in the source in order to build sile.